charms/volcano-scheduler/src/manifests.py

```diff
@@ -0,0 +1,127 @@
+"""Apply extra manifests for enabling the scheduler and its config."""
+
+import logging
+from pathlib import Path
+from typing import List, Sequence
+
+from jinja2 import Environment, FileSystemLoader
+from lightkube import Client, codecs
+from lightkube.core.exceptions import ApiError
+from lightkube.core.resource import Resource
+from lightkube.generic_resource import load_in_cluster_generic_resources
+from lightkube.resources.apps_v1 import StatefulSet
+
+log = logging.getLogger(__name__)
+CRD_BASE = "v1"  # assumes we're in a k8s cluster that has access to v1 CRDs
+
+
+class Manifests:
+    """Render manifests from charm config and apply to the cluster."""
+
+    def __init__(self, charm):
+        self._charm = charm
+        self.namespace = charm.model.name
+        self.application = charm.app.name
+        self.client = Client(namespace=self.namespace, field_manager=self.application)
+        load_in_cluster_generic_resources(self.client)
+
+    @property
+    def _resources(self) -> Sequence[Resource]:
+        templates = Path("templates")
+        templates = (templates / "scheduler.yaml", *(templates / "crd" / CRD_BASE).glob("*.yaml"))
```

Comment:
Do we need anything from scheduler.yaml? I see:

1. a ClusterRole, which is not needed because we're relying on `--trust` to apply it anyway
2. a ClusterRoleBinding, which is not needed for the same reason (and binds to a ServiceAccount named `volcano-scheduler`, which could be wrong if the application name is different?)
3. a Service, which shouldn't be needed if we can use Juju's new `open-port` support for k8s charms to open ports on the service provided by Juju
4. Other stuff that is trimmed away because Values.juju.scheduler=true (makes sense!)

It looks to me like we don't need to apply anything from there. If possible, I would just remove this manifest entirely.